src/ripple/protocol/impl/PublicKey.cpp

```diff
@@ -179,23 +181,25 @@ ed25519Canonical (Slice const& sig)
 
 PublicKey::PublicKey (Slice const& slice)
 {
-    if(! publicKeyType(slice))
+    if (slice.size() != buf_.size() || !publicKeyType(slice))
```

Comment:
The size check here is redundant: `publicKeyType` does it for you and will return an unseated `boost::optional` if the size isn't right (i.e. exactly 33 bytes).

Reply:
Addressed in bcaaf38a3e0e9516fd5ac28d73a7721de73188c0

Note: I understood that the check was redundant. I made the change because I had initially thought a key could 32 or 33 bytes and if I saw code like this in the constructor it would have helped me. However, given the change removing the `size_` (and as you point out the code in `publicKeyType`) it very clear the key is always 33 bytes. Removed the check.